tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerTests.cs

```diff
@@ -753,10 +766,18 @@ public static bool IsRecovering(this Server server, ServerErrorCode code, string
             return (bool)methodInfo.Invoke(server, new object[] { code, message });
         }
 
-        public static bool ShouldInvalidateServer(this Server server, Exception exception)
+        public static bool ShouldInvalidateServer(this Server server,
+            IConnectionHandle connection,
+            Exception exception,
+            ServerDescription description,
+            out TopologyDescription? responseTopologyVersion)
         {
             var methodInfo = typeof(Server).GetMethod(nameof(ShouldInvalidateServer), BindingFlags.NonPublic | BindingFlags.Instance);
-            return (bool)methodInfo.Invoke(server, new object[] { exception });
+            var parameters = new object[] {connection, exception, description, null};
+            int outParameterIndex = Array.IndexOf(parameters, null);
```

Comment:
The index in the parameter array where we've placed "null" is the index at which the out parameter's value will be written.